### DISCUSSION TITLE
Dynamically spawned physics objects with collision

### DISCUSSION THREAD

**Discussion by mkremins** [mozilla/hubs#4104] [Hubs Cloud Questions] [answered]
Hi! We're building a custom hubs-cloud client and we'd like to dynamically spawn physics objects that collide with the ground and with one another. We're currently spawning physics objects as follows:

```js
const ent = document.createElement("a-entity");
ent.setAttribute("networked", "template:#spawnedObj-drawing");
ent.setAttribute("body-helper", {
  collisionFilterMask: -1,
  gravity: {x: 0, y: -9.8, z: 0}
});
APP.scene.appendChild(ent);
```

...using the following template:

```html
<template id="spawnedObj-drawing">
  <a-entity class="spawnedObj" geometry="primitive:sphere"></a-entity>
</template>
```

...but these spawned objects fall through the environment and don't collide with one another, even though `-1` is the constant specified for `COLLISION_LAYERS.ALL` in [`constants.js`](https://github.com/mozilla/hubs/blob/hubs-cloud/src/constants.js#L4).

Are we missing something about how the `collisionFilterMask` parameter to the `body-helper` is supposed to work, or looking in the wrong place entirely? Do we need to somehow manually specify a collider shape/size for our spawned objects? Does the fact that the objects are networked interfere with the physics behavior somehow?

Thanks in advance!

**Comment by johnshaughnessy** [comment, marked as answer]
> Do we need to somehow manually specify a collider shape/size for our spawned objects? 

Yes, you will need `body-helper` and `shape-helper` along with probably several more things to get the objects to behave as intended. I don't have any documentation on the physics system to point you to. The best resource we have for this is guide: [Creating New Networked Interactables](https://github.com/mozilla/hubs/blob/f7a74cab8ba6babdbe188d7755ce46764fc1bac3/doc/creating-networked-interactables.md#creating-new-networked-interactables)

**Reply by mkremins** [reply]
Awesome, thanks! `shape-helper` was exactly the thing we were looking for.
